### DISCUSSION TITLE
【柴柴小知識】關於Azure佈署- 跑到哪了？到底要等多久？

### DISCUSSION THREAD

**Discussion by wolfyeva** [1101-GS3073-C/Python-Community-GS3073-C#329] [General]
# 問題
## 看不出來他到底有沒有在跑，有沒有出錯....
當我們在使用Azure 佈署，連結Github存放庫設定完成後，點入網站只能看到「Your web app is running and waiting for your content」，還寫了佈署後還需要等大概5分鐘。
![螢幕擷取畫面 2022-05-23 000855](https://user-images.githubusercontent.com/81672797/169706333-3f591af2-e2d1-47e7-940d-68cd8e04cd3e.png)
看著這個頁面重複F5刷新，**完全看不出來他到底有沒有在跑，有沒有出錯....**

# 解決
## 步驟一：點選「佈署中心」->「分支」點選連結
![Inked螢幕擷取畫面 2022-05-23 003425_LI](https://user-images.githubusercontent.com/81672797/169706521-7f3fb513-df38-4ee2-a5e4-08f35459d903.jpg)
將會跳出此Github存放庫的頁面。
## 步驟二：點選「Action」
![螢幕擷取畫面 2022-05-23 003310](https://user-images.githubusercontent.com/81672797/169706617-5a3531cd-95f7-41d4-affe-67af3f26bb0d.png)
**這裡就可以觀察佈署狀況！** 下圖為這個workflow的運作狀況！
![螢幕擷取畫面 2022-05-23 001040](https://user-images.githubusercontent.com/81672797/169706680-eb542165-e8b8-45c9-b07e-73aad30c48ba.png)

### 狀況一：成功
如果向下圖為綠燈，就代表佈署成功了！
![螢幕擷取畫面 2022-05-23 001114](https://user-images.githubusercontent.com/81672797/169706720-bb787566-12a9-49b5-8408-640a4e311b12.png)
這個範例共花了2分26秒完成佈署！

### 狀況二：失敗
![螢幕擷取畫面 2022-05-23 010131](https://user-images.githubusercontent.com/81672797/169706884-5393b296-4de2-4c0c-bb70-f807dd449940.png)
知道佈署失敗，並且從錯誤訊息中得知忘記放requirements.txt 導致build 失敗。

## 結論
可以透過Github Action 得知佈署進度，以及錯誤訊息！

**Comment by lx3211** [comment]
謝謝分享！不久後就會用到！

**Comment by NcuJerryKao** [comment]
感謝你 這問題無疑是 替大家鋪平了前路

**Comment by 32844583** [comment]
謝謝分享，還真的沒發現有這種功能呢~

**Comment by Ben6uo** [comment]
謝謝分享 我第一次知道Azure可以這樣看
